mk/waf.mk

```diff
@@ -95,12 +95,19 @@ all:
 	@echo BINARY_NAME=$(BINARY_NAME)$(BINARY_FRAME)
 	@echo BINARY_EXTENSION=$(BINARY_EXTENSION)
 	@echo BINARY_EXTENSION_DEST=$(BINARY_EXTENSION_DEST)
+	mkdir -p $(SKETCHBOOK)/build
 	cd $(SKETCHBOOK) && modules/waf/waf-light --board $(TARGET_BOARD) configure
 	cd $(SKETCHBOOK) && modules/waf/waf-light $(TARGET_BUILD) $(WAF_FLAGS)
 	@cp $(SKETCHBOOK)/build/$(BIN_DIR)/bin/$(BINARY_NAME)$(BINARY_FRAME)$(BINARY_EXTENSION) $(SKETCH)$(BINARY_EXTENSION_DEST)
 	@ls -l $(SKETCH)$(BINARY_EXTENSION_DEST)
 	@echo "Build done $(SKETCH)$(BINARY_EXTENSION_DEST)"
 
 clean:
-	@cd $(SKETCHBOOK) && modules/waf/waf-light clean
+	mkdir -p $(SKETCHBOOK)/build
+	@cd $(SKETCHBOOK) && modules/waf/waf-light --board $(TARGET_BOARD) configure clean
```

Comment:
I see some problems:
 - How would a clean command be issued?
     - `make clean` would only work for `sitl`.
     - `make <board-spec> clean` or `make clean <board-spec>` would trigger a build command too, right?
 - We're awkwardly forcing a `configure` command in order to be able to run `clean` with no errors.
 - As pointed above, this only works for `sitl`. But even if it worked for other boards, it would be missing debug variants.

With that said, I think the key reason for those problems is the fact that the old make-base build system didn't have the concept of a configuration step. I think this `clean` target should just call Waf's `distclean` command. That's the behavior that is closest to what the old build system did and we don't need workarounds like these.